base36.go

```diff
@@ -0,0 +1,158 @@
+/*
+
+Package base36 provides a reasonably fast implementation of a binary base36 codec.
+
+*/
+package base36
+
+// Simplified code based on https://godoc.org/github.com/mr-tron/base58
+// which in turn is based on https://github.com/trezor/trezor-crypto/commit/89a7d7797b806fac
+
+import (
+	"fmt"
+	"math"
+)
+
+const UcAlphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const LcAlphabet = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+var encExpansion int
+var revAlphabet [256]uint64
```

Comment:
Why not map to bytes? Or uint8?

Reply:
Because we use the value [only once doing uint64 math](https://github.com/multiformats/go-base36/blob/11f3b26a8e0e9068d38d6fa365d487dd00c6d057/base36.go#L126-L130), so might as well just keep it like this in memory.

Reply:
Yeah, but smaller values might fit better into the CPU cache. If we're going to go there :).

Also, why not do the math with smaller values? I've noticed that intel chips perform better with smaller types (usually).

Reply:
I do not fully understand the algorightm tbh, but with 32bits it overflows ( tests fail ). Let me fiddle with it a bit more...

Reply:
Doesn't make a difference on benches, left it at []byte ( plus much smaller table ):
https://github.com/multiformats/go-base36/blob/438fc2b1a7f50241be57e5d9801441c5050fa9fa/base36.go#L17-L20